include/seqan3/alignment/configuration/align_config_band.hpp

```diff
@@ -14,51 +14,104 @@
+ * by the lower and the upper diagonal.
+ *
+ * If this configuration is default constructed or not set during the algorithm configuration the the full alignment
+ * matrix will be computed.
+ *
+ * During the construction of this configuration element or before the execution of the alignment algorithm the
+ * band configuration is validated. If the user provided an invalid band, e.g. the upper diagonal is smaller than
+ * the lower bound, or the alignment matrix would be ill configured such that the requested alignment method cannot
+ * be computed (because the global alignment requires the first cell and the last cell of the matrix to be reachable),
+ * then a seqan3::invalid_alignment_configuration will be thrown.
  *
  * ### Example
  *
  * \include test/snippet/alignment/configuration/align_cfg_band_example.cpp
  */
-template <typename band_t>
-//!\cond
-    requires std::same_as<band_t, static_band>
-//!\endcond
-struct band : public pipeable_config_element<band<band_t>, band_t>
+class band_fixed_size : public pipeable_config_element<band_fixed_size, seqan3::detail::empty_type>
 {
+private:
+    //!\brief The base class type.
+    using base_t = pipeable_config_element<band_fixed_size, seqan3::detail::empty_type>;
+public:
+
+    //!\brief The selected lower diagonal. Defaults to 0.
+    seqan3::align_cfg::lower_diagonal lower_diagonal{std::numeric_limits<int32_t>::lowest()};
+    //!\brief The selected upper diagonal. Defaults to 0.
+    seqan3::align_cfg::upper_diagonal upper_diagonal{std::numeric_limits<int32_t>::max()};
+
+    constexpr band_fixed_size() = default; //!< Defaulted.
+    constexpr band_fixed_size(band_fixed_size const &) = default; //!< Defaulted.
+    constexpr band_fixed_size(band_fixed_size &&) = default; //!< Defaulted.
+    constexpr band_fixed_size & operator=(band_fixed_size const &) = default; //!< Defaulted.
+    constexpr band_fixed_size & operator=(band_fixed_size &&) = default; //!< Defaulted.
+    ~band_fixed_size() = default; //!< Defaulted.
+
+    /*!\brief Initialises the fixed size band by setting the lower and the upper matrix diagonal.
+     *
+     * \param lower_diagonal \copybrief seqan3::align_cfg::band_fixed_size::lower_diagonal
+     * \param upper_diagonal \copybrief seqan3::align_cfg::band_fixed_size::upper_diagonal
+     *
+     * \details
+     *
+     * The lower diagonal represents the lower bound of the banded matrix, i.e. the alignment cannot pass below this
+     * diagonal. Similar, the upper diagonal represents the upper bound of the alignment. If the upper diagonal is
+     * smaller than the lower diagonal an exception will be thrown.
+     *
+     * \throws seqan3::invalid_alignment_configuration
+     */
+    constexpr band_fixed_size(seqan3::align_cfg::lower_diagonal const lower_diagonal,
+                              seqan3::align_cfg::upper_diagonal const upper_diagonal) :
+        lower_diagonal{std::move(lower_diagonal)},
```

Comment:
Is this safe considering the names?

Reply:
Yes, since the types are fully qualified. This was necessary to make this happen:
https://godbolt.org/z/P_BbSA